### PR TITLE
update readme to include info about all possible commands and options

### LINKED ISSUE #153: Update info in the README
The README doesn't contain all the info about all the possible oghliner commands, we should update it.

### PR DESCRIPTION
**Pull request by mykmelez**
Fixes #153.

### CODE REVIEW

**Comment by mykmelez** [comment]
This is now ready for review. You'll probably be better off reviewing the [new rendered version of the README](https://github.com/mykmelez/oghliner/blob/update-readme/README.md) rather than the patch itself, since I rewrote most of the file

**Comment by marco-c** [comment]
I know you haven't touched this part, but it needs to be updated as well:

``` js
 var oghliner = require('oghliner');

 gulp.task('offline', function(callback) {
   oghliner.offline({
     rootDir: 'dist/',
     fileGlobs: [
       '**/*.html',
       'js/**/*.js',
     ],
   }, callback);
 });

 gulp.task('deploy', function(callback) {
   oghliner.deploy({
     rootDir: 'dist/',
   }, callback);
 });
```

The `offline` and `deploy` no longer have a callback as the second argument, so we should remove it from the example.
Gulp supports promises, so you can simply modify it to:

``` js
 var oghliner = require('oghliner');

 gulp.task('offline', function() {
   return oghliner.offline({
     rootDir: 'dist/',
     fileGlobs: [
       '**/*.html',
       'js/**/*.js',
     ],
   });
 });

 gulp.task('deploy', function() {
   return oghliner.deploy({
     rootDir: 'dist/',
   });
 });
```

File: README.md [hunk 1]
```diff
@@ -1,56 +1,316 @@
...
+
+Then you can offline and deploy the app using the offline and deploy commands.
+
+ℹ For more information about offlining and deployment, see:
+    https://mozilla.github.io/oghliner/
+```
+
+Then add a &lt;script> tag referencing that script to your app's HTML page(s)/template(s). Oghliner will suggest one, but it doesn't necessarily know your app's directory structure, so make sure it contains the correct path to the file!
+
+## Offline
```
Comment (marco-c) [inline README.md:165]: I'd specify the `offline` and `deploy` return type (promise), it could be useful for developers.

**Comment by marco-c** [comment]
Looks great!

**Comment by mykmelez** [comment]
> I know you haven't touched this part, but it needs to be updated as well:

Ah, indeed, updated in 846fc43.

File: README.md (hunk 1 above)
Comment (mykmelez) [inline README.md:165]: > I'd specify the offline and deploy return type (promise), it could be useful for developers.

Good idea, documented in 82cc1af.

**Pull request merged**